Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,151 @@
+
+Until that is implemented, we could alternatively monitor mutations in Current by counting the number of times Current is changed in the `setup()` and `tearDown()` methods. We would have to whitelist every test file with the exception of snapshot tests (see link above) using the regex `\b(?!Snapshot)(.*Tests)` and then use Danger to manually parse the test line-by-line.
+
+1) Find the `setUp()` function inside the test file. If found, we would determine the boundaries of this function (by counting each time an open curly bracket appears and doing the reverse for closing curly brackets) and we'd use the following regexes to detect changes in Current: 
+    a) `Current\..*\ = `.
+    b) `Current.changeTheme\(.*`
+    c) `Current.set\(.*`
+
+2) Repeat step 1 but for `tearDown()` instead.
+
+3) Assess if the number of changes inside each function matches.
+    a) if so, then no issue would be found
+    b) if not, then Current appeared to have been mutated but not restored; we'd issue a warning for the lines in which the algorithm detected a change and alert the developers in GitHub using Danger.
+
+Nevertheless, this rule is rather naïve because we have no actual way of detecting if Current was reset or not; it would merely indicate that in all likeliness the developer has forgotten to restore Current since the number of changes in both methods does not match.
+Additionally, there are plenty of cases through developers could modify Current and easily side-step this rule (eg: if the mutations and resets were made in auxiliar methods instead).
+
+As such, this approach is not a silver bullet or anything remotely similar; such a feature would require compiling and running the actual Swift code, not to mention that we'd have to make Current conform to Equatable.
+
+
+### Translation errors
+
+Translations are particularly hard to catch during PR code reviews, especially if a lot of new phrases are added at once. While, in theory, we have a fastlane set up to deal with these translations there is always the risk that developers mistype the localization key when setting up a new static var.
+
+As such, I'd like to suggest another rule for Danger that would:
+    1. detect new static String variables in every Localizable.swift file.
+    2. if we detected a new static variable we would then scan the localization.string files and look for that key; if we didn't find them that would mean that there was probably a typo and the actual translation would never  appear on-screen (only the key itself).
+
+Furthermore, the reverse situation would also be problematic (albeit at a different scale): a key in the `.strings` file without an equivalent constant in its Swift counterpart would likely mean that the developer had added unused strings to the project. This would also be detectable using Danger.
+
+We would need to whitelist `Localizable.*\.swift` as well as the actual `Localizable.string` files in order to implement both rules.
```

Comment:
let's not forget the `TargetSpecificLocalizable.strings`

Reply:
also let's not forget what @dmcrodrigues mentioned: SwiftGen to do not rely on developers

Reply:
@daniloaliberti @dmcrodrigues I added some tweaks regarding the localization rule, please have a look when you've time